zerver/actions/custom_profile_fields.py

```diff
@@ -209,6 +215,48 @@ def do_update_user_custom_profile_data_if_changed(
             },
         )
 
+        new_value = get_custom_profile_field_display_value(field_value)
+
+        changes.append(
+            UserProfileChangeDict(
+                field_name=field_value.field.name,
+                old_value=old_value,
+                new_value=new_value,
+            )
+        )
+
+    if changes and acting_user is not None:
+        send_user_profile_update_notification(
+            user_profile=user_profile, acting_user=acting_user, changes=changes
+        )
+
+
+def get_custom_profile_field_display_value(field_value: CustomProfileFieldValue) -> str:
+    """Convert custom profile field value to human-readable string based on field type."""
```

Comment:
Do we really not have logic like this in the server already? Answer might be no.